iidm/iidm-extensions/src/main/java/com/powsybl/iidm/network/extensions/ActivePowerControl.java

```diff
@@ -25,8 +25,66 @@ default String getName() {
 
     void setParticipate(boolean participate);
 
+    /**
+     * This is the change in generator power output divided by the change in frequency
+     * normalized by the nominal power of the generator and the nominal frequency and
+     * expressed in percent and negated. A positive value of speed change droop provides
+     * additional generator output upon a drop in frequency.
+     * @return Governor Speed Changer Droop.
+     */
     float getDroop();
 
+    /**
+     * @param droop new Governor Speed Changer Droop value
+     */
     void setDroop(float droop);
 
+    /**
+     * The sum of the participation factors across generating units does not have to sum to one.
+     * It is used for representing distributed slack participation factor.
+     * The attribute shall be a positive value or zero.
+     * @return Generating unit short term economic participation factor.
+     */
+    float getShortPF();
+
+    /**
+     * @param shortPF new short term economic participation factor value
+     */
+    void setShortPF(float shortPF);
+
+    /**
+     * The sum of the participation factors across generating units does not have to sum to one.
+     * It is used for representing distributed slack participation factor.
+     * The attribute shall be a positive value or zero.
+     * @return Generating unit economic participation factor.
+     */
+    float getNormalPF();
+
+    void setNormalPF(float normalPF);
+
+    /**
+     * The sum of the participation factors across generating units does not have to sum to one.
+     * It is used for representing distributed slack participation factor.
+     * The attribute shall be a positive value or zero.
+     * @return Generating unit long term economic participation factor.
+     */
+    float getLongPF();
+
+    /**
+     * @param longPF new long term economic participation factor value
+     */
+    void setLongPF(float longPF);
+
+    /**
+     * Priority for use as powerflow voltage phase angle reference bus selection.
+     * 0 = don't care (default) 1 = highest priority. 2 is less than 1 and so on.
+     * @return reference priority
+     */
+    int getReferencePriority();
```

Comment:
We already have an extension SlackTerminal for that (see CGMES import how referencePriority is used to create a SlackTerminal extension)

Reply:
SlackTerminal could indeed work for this but not in the general case:
- there can be only one SlackTerminal per VoltageLevel. In the case where there are two generators in on VoltageLevel, gen1 with priority 1, gen2 with priority 2. Say both are connected initially, but then the user disconnects the generator in priority 1, the other generator with priority 2 cannot be picked up.
- SlackTerminal-s across different VoltageLevel-s do not have have any relative priority. So If I have two VoltageLevel-s, VL1 with gen1 of referencePriority1, VL2 with gen2 of referencePriority2, in this case it is not possible to represent that gen1 has higher priority than gen2 for slack slelection.

Reply:
The problem with referencePriority on generators is that it does not cover the general case of slack bus  selection but only slack bus on PV buses. Only very old power flow, does not allow general slack bus selection and most of the time best slack bus is not on a PV bus (like when you select most meshed bus). Also we tend to have more and more generators modelling on MT in our cases (so in 15Kv behind a step up transformer) and having a slack of this kind of bus is not robust.

Maybe we should just try to improve/enrich existing SlackTerminal extension in order to meet your needs ?

Reply:
just thinking... an alternative solution could be to enhance the existing SlackTerminal extension to allow 1/ for multiple SlackTerminal extensions to be present in a given VoltageLevel, and 2/ add priority to the extension. But This would cause issues for backward compatibility of existing tools relying on this extension, and 2/ the very same extension is used for slack selection results, and when it comes to slack selection results, priority don't really make sense any more.

Reply:
I de-scoped referencePriority from this PR, have to think about it

Reply:
Why not, maybe in order to complete this PR, we can just remove the referencePriority and think about how to improve slack selection part in another issue/PR ?

Reply:
I removed referencePriority from this PR and entered issue #2412 to track this topic